src/main.ts

```diff
@@ -0,0 +1,113 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as fs from 'fs';
+import { request } from '@octokit/request';
+import { Octokit } from "@octokit/core";
+fs.readdir('./', (err, files) => {
+  console.log("entered");
+  if (err)
+    console.log(err);
+  else {
+
+    //readme checks
+    readmeChecks(files);
+
+    //code owners in .github
+    codeownerCheck();
+
+    //dont have node modules in master for .ts
+    nodemodulesCheck();
+
+    //check pr protection in branch
+    start();
+    
+    }})
+
+    function getExtension(filename: string) {
+      return filename.substring(filename.lastIndexOf('.')+1, filename.length)   
+    }
+
+    function readmeChecks(files: string[]){
+
+      const includesReadme = files.includes('README.md');
+      if(includesReadme){
+        console.log("Found readme");
+        fs.readFile('./README.md', function (err, data) {
+        //check example
+        if(data.includes('Example'))
+          console.log("found example");
+        else
+          console.log("Example not found");
```

Comment:
The error messages should be more intuitive. Something like :
"Please add example workflow in the readMe."